gitpod-io/gitpod
# Workspace integration tests: Poll for job phase instead of tailing logs

## Description


The workspace integration tests have been failing consistently since we switched to gRPC with an error related to communicating with Werft (see [issue](https://github.com/gitpod-io/gitpod/issues/11979)).

I narrowed the error down to when we use `werft job logs` and was able to get that to fail consistently in Werft by triggering the integration test jobs. However, I wasn't able to get `werft job logs` to fail when using the command from a Gitpod workspace.

So instead of trying to modify any timeouts in our system I opted to instead change how the integration test job waits for the completion of the build job for the preview environment. We now poll for that phase until the job completes and then check if the job was successful. The downside of this approach is that you no longer get the logs of the build job in your integration tests jobs but I think that's a reasonable trade-off.

I decided to move your bash script into a separate file as I found it hard to edit the bash inside of YAML. If you're unhappy with this change let me know and I can move it back. Because of that change this PR is easiest to read commit-by-commit ☺️

I also cleaned up some of the output of the script a bit so that the logs goes into the relevant slices (see the difference between [gitpod-custom-mads-ws-integration-tests.1](https://werft.gitpod-dev.com/job/gitpod-custom-mads-ws-integration-tests.1) and [gitpod-custom-mads-ws-integration-tests.2](https://werft.gitpod-dev.com/job/gitpod-custom-mads-ws-integration-tests.2)

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/11979

## How to test


During development I ran this job using:

```
werft job run github -j .werft/workspace-run-integration-tests.yaml -s .werft/workspace-run-integration-tests.sh
```

See the most recent [job here](https://werft.gitpod-dev.com/job/gitpod-custom-mads-ws-integration-tests.2). The integration tests don't pass but I think that's outside the scope of this PR - at least it's able to run the tests again with this change ☺️

## Release Notes

```release-note
NONE
```

## Documentation


N/A

## Werft options:
